### DISCUSSION TITLE
Integration keeps showing the last known value for solar production after plant offline

### DISCUSSION THREAD

**Discussion by terppatyyppi** [hultenvp/solis-sensor#204] [Q&A]
I've been using this now for a total of 1 day, hence I'm hoping this is due to something I've done wrong...but here's the situation:
1. Integration works now with SolisCloud
2. Sensors seem reliable while the plant is up and running
3. Once the inverter goes offline in the evening, the value of the current power output seems to stay at the last known value, for me 20W yesterday evening.
4. So, when there is no power generation at all, this keeps showing 20W output until the plant comes online again in the morning, and also the "animation" flowing in the solar generation part of the Energy dashboard.

Any idea on how to prevent this from happening?

**Comment by hultenvp** [comment]
This is odd, I do not see the same behaviour in my setup. Does your system have a battery installed?

**Reply by terppatyyppi** [reply]
No, no battery installed. I've heard someone else reporting this same behaviour.

**Reply by hultenvp** [reply]
Yes, but I recall only seeing this on energy today, which has indeed reset issues on m.ginlong.com

**Reply by terppatyyppi** [reply]
Might be.
Anyway, this last known value sticks with the Solis sensor itself. Mine is continously showing 20W output through the nights.

**Comment by terppatyyppi** [comment]
I've been playing with this so much, that I never saw another problem that I have with this (so many HA restarts per day that it never revealed itself until now):

In addition to the sensor never going to zero when plant offline, the integration never works the next day until I reload the integration or restart HA (i.e. never updates the sensor values from that said 20W output from previous evening that stays on after plant going offline).

So the only way to keep this up and running, is to manually reload it every day once the plant is online again. 

What could be causing this type of issues?

**Reply by hultenvp** [reply]
That could very well be a reason. I do assume you run the latest version? I fixed the retry issue in v 3.1.0. That one caused the integration to stop working if there was a read error from the server.

**Reply by terppatyyppi** [reply]
It seems I was still running 3.0.9, but updated now. Will know tomorrow if it helped.

**Reply by hultenvp** [reply]
For me this issue was also hidden on my test VM as it is running on my dev laptop. I could reproduce on "production" and fixed it there. Now running pretty stable on 3.1.0, so I have good hopes it will help you. If still issues, please share the log

**Reply by terppatyyppi** [reply]
I can confirm this morning it works without a reload!

Now the only remaining thing is to figure out why the sensor shows 20W output (and not 0, or at minimum unavailable) when the inverter is offline.

**Comment by supergregg** [comment]
I created an bug report, as I missed this discussion. I have the same problem, output showing 20W all night. The same value is visible in the solis cloud app if you look around when the inverter is offline. This is probably a quirk of the API, but the app manages this, probably by setting it to zero if the inverter is offline. The API is probably just returning last known value untill the value isuppdated from the inverter. Values of voltages and frequency experience the same problem.

**Reply by hultenvp** [reply]
@supergregg , @terppatyyppi This looks like the same issue as with "energy today". There also the last measured value is reported while the inverter is off. Apparently this is also the case for power output.
I'm in the middle of a home renovation, so do not have much time to work on the integration. Please keep an eye on the issue supergregg posted.

**Comment by turlos** [comment]
Any updates on this? I have the same issue, the fake 20w during the night also appears to be combined and added to the energy dashboard at the point the inverter comes back online, skewing 'Generated today' measurements.
![image](https://github.com/hultenvp/solis-sensor/assets/10931507/868c01ef-625b-4c8a-a96d-d06e0bda45c1)

**Reply by terppatyyppi** [reply]
Something has happened quite recently, I believe. My integration has now (finally) started showing 0 (zero) during the inverter being off. Good!

**Reply by hultenvp** [reply]
There were some fixes dropped in 3.4.0 and 3.4.1 to filter out nonsense coming from the API.
Hope this catches all corner cases
